Workflow/Workflow_FailoverFailback.ps1

```diff
@@ -0,0 +1,232 @@
+
+Write-Output ">>> Connect to a tintri server $source_tintriserver.`n"
+
+$source_ts = Connect-TintriServer -Server $source_tintriserver -UserName $source_tsusername -Password $source_tspassword -SetDefaultServer
+$source_ts
+
+
+Write-Output ">>> Connect to a tintri server $target_tintriserver.`n"
+
+$target_ts = Connect-TintriServer -Server $target_tintriserver -UserName $target_tsusername -Password $target_tspassword
+$target_ts
+
+
+Write-Output ">>> Connect to a VI server $viserver.`n"
+
+if ($global:DefaultVIServers) { Disconnect-VIServer * -Force -confirm:$false -ea SilentlyContinue | out-null }
+Connect-VIServer -Server $viserver -User $viusername -Password $vipassword
+
+
+Write-Output ">>> Get the tintri vm $vmname on tintri server $source_tintriserver.`n"
+
+$source_vm = Get-TintriVM -Name $vmname -TintriServer $source_ts
+$source_vm
+
+
+Write-Output ">>> Get the datastore repl path for tintri server $source_tintriserver.`n"
+
+$source_rp = Get-TintriDatastoreReplPath -HostName $source_ts.ApplianceHostName
+$source_rp
+
+Write-Output ">>> Replicate vm $vmname from tintri server $source_tintriserver to $target_tintriserver.`n"
+
+New-TintriVMReplConfiguration -VM $source_vm -DatastoreReplPath $source_rp
+
+
+Write-Output ">>> Make sure the replication is completed. Use 'Get-TintriVM' to refresh the VMStore.`n"
+
+do {
+
+    $vmlist = Get-TintriVM -TintriServer $source_ts
+
+    $latest_ss = (Get-TintriVM -Name $vmname -TintriServer $source_ts).Snapshot.Latest.Uuid.UuId
+
+    $latest_repl = (Get-TintriVM -Name $vmname -TintriServer $source_ts).Snapshot.LatestReplicated.Uuid.UuId
+
+    $rpCfg = Get-TintriVMReplConfiguration -VM $source_vm
+    
+    if (($rpCfg.ReplicationState -ne "RUNNING") -or ($rpCfg.IsEnabled -eq $false))
+    {
+        Write-Error "Replication is not configured please do so and try again.`n"
+        return
+    }
+	
+	Start-Sleep -Seconds 1
+	Write-Output "Waiting for replicaiton snapshots: $latest_ss ...to equal... $latest_repl"
+}
+while (($latest_ss -ne $null) -and ( $latest_ss -ne $latest_repl ))
+
+
+Write-Output ">>> Remove the replication for vm $vmname on tintri server $source_tintriserver.`n"
+
+Remove-TintriVMReplConfiguration -VM $source_vm
+
+
+Write-Output ">>> Delete vm $vmname from tintri server $source_tintriserver.`n"
+
+$vm = Get-VM -Name $vmname
+
+Stop-VM -VM $vm -Confirm:$False
+
+
+Remove-VM -DeletePermanently -VM $vm -Confirm:$false
+
+
+Write-Output ">>> Run 'Get-TintriVM' to refresh the tintri server $target_tintriserver.`n"
+
+Get-TintriVM -TintriServer $target_ts
+
+
+Write-Output ">>> Restore vm $vmname on tintri server $target_tintriserver.`n"
+
+Restore-TintriVM -Name $vmname -TintriServer $target_ts -DestinationDirectory $target_destination_directory -UseLatestSnapshot
+
+
+Write-Output ">>> Add vm $vmname on tintri server $target_tintriserver to inventory.`n"
+
+$target_vmfilepath = '[' + $target_ts.HostNameOrIp + '] ' + $target_destination_directory + '/' + $vmname + '.vmx'

```

Comment:
-VMFilePath has format: '[datastore] path', so we should replace here with something like: 
$target_datastore = Get-TintriHypervisorDatastore -TintriServer $target_ts

$target_vmfilepath = '[' + $target_datastore.DisplayName + '] ' + $target_destination_directory + '/' + $vmname + '.vmx'

New-VM -VMFilePath $target_vmfilepath -VMHost $target_esx